src/bsk201909/secrets.md

```diff
@@ -0,0 +1,264 @@
+## How do you securely manage secrets in Kubernetes?
+
+Kubernetes secrets hold the most sensitive information of your application - API keys, tokens, database passwords, etc.
+
+If a hacker can retrieve one of these secrets, they could connect to your database without you even noticing it.
+
+It's important to ensure that those secrets are stored as securely as possible.
+
+Let's recap on how secrets work in Kubernetes.
+
+Secrets are objects that contain key-value pairs, and some metadata.
+
+Secrets are similar to ConfigMaps and share the same limitations (1MB in size — as an example).
+
+The main differences are the security protections added by Kubernetes teams to secrets.
+
+This is how a secret looks like in a YAML representation:
```

Comment:
Base64 encoding is not specific to Secretes, it's also used in ConfigMaps, so people should know it already. So I would refactor the following block to something like this:

`````markdown
This is how a secret looks like in a YAML representation:

```yaml
...
```

Just like a ConfigMap, it has a `data` field containing the key-value pairs that make up the stored data.

Also like in a ConfigMap, the values of the key-value pairs are Base64 encoded - this is to allow for the representation of binary data.

> [Base64](https://en.wikipedia.org/wiki/Base64) is an encoding that can represent any type of data as ASCII strings. It should not be confused with an encryption.

You can create a secret like any other Kubernetes object (usually with kubectl).

The manifest will be sent to the server that will proceed it and store it in etcd - similarly to all Kubernetes objects.

But we still haven't answered how to properly secure those secrets.

...
`````

Reply:
I just checked the [docs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-a-configmap) again to be 100% sure, ConfigMap is not base64 encoded because it's not intended to hold binary data.

Reply:
Oops, that's true :)